test/end2end_test.go

```diff
@@ -6994,11 +7031,10 @@ func (s *httpServer) start(t *testing.T, lis net.Listener) {
 	}()
 }
 
-func doHTTPHeaderTest(t *testing.T, errCode codes.Code, headerFields ...[]string) {
-	t.Helper()
+func doHTTPHeaderTest(t *testing.T, errCode codes.Code, headerFields ...[]string) error {
 	lis, err := net.Listen("tcp", "localhost:0")
 	if err != nil {
-		t.Fatalf("Failed to listen. Err: %v", err)
+		return fmt.Errorf("failed to listen. Err: %v", err)
```

Comment:
I have started making this comment very recently that we should try to make out error strings concise, but convey all the important information that it needs to contain. This means that we should try to avoid filler words like failed, error etc. 

So, I would recommend changing the error strings in this function as follows:

`return fmt.Errorf("listening on %q: %v", "localhost:0", err)`

`return fmt.Errorf("dial(%q): %v", lis.Addr().String(), err)`

`return fmt.Errorf("creating FullDuplex stream: %v", err)`

`return fmt.Errorf("stream.Recv() = %v, want error code: %v", err, errCode)`

Reply:
done!